Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,16 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    system "pkill postgres || true"
+
+    if (var/"postgres").exist?
+      mv var/"postgres", var/"postgres-#{version_suffix}"
+    elsif (var/"postgres-#{version_suffix}").exist?
+      puts ""
```

Comment:
Seems you can just remove this logic if the situation doesn't need any logic of its own? Also, as is the section below will never be called as the `and`ed ifs there are covered by the those `if`s above.